low-level/files/opt/seagate/sspl/bin/feed_sspl_conf_to_consul.py

```diff
@@ -0,0 +1,240 @@
+# cortx-questions@seagate.com.
+
+"""
+ ******************************************************************************
+  Description:   Parse config file and arguments to set common & SSPL config
+  Purpose    :   To feed sspl and common configuration to consul
+  Usage      :   python3 feed_sspl_conf_to_consul.py [-F] [-C] [-N] [-Ru] [-Rp]
+ ******************************************************************************
+"""
+
+import re
+import os
+import sys
+import consul
+import argparse
+from configobj import ConfigObj
+
+from cortx.utils.process import SimpleProcess
+
+CONSUL_HOST = '127.0.0.1'
+CONSUL_PORT = '8500'
+
+
+class LoadConfig:
+
+    def __init__(self):
+        """ Establish consul connection
+        """
+        host = os.getenv('CONSUL_HOST', CONSUL_HOST)
+        port = os.getenv('CONSUL_PORT', CONSUL_PORT)
+        self.consul_conn = consul.Consul(host=host, port=port)
+        self.rmq_same_pass_sect = ["LOGGINGPROCESSOR",
+                                   "RABBITMQEGRESSPROCESSOR",
+                                   "RABBITMQINGRESSPROCESSOR"]
+        self.common_config_sect = ['system_information',
+                                   'storage_enclosure',
+                                   'bmc',
+                                   'rabbitmq']
+        # TODO: Get persistent data from ConfigStore or cortx-utils
+        self.persistent_data_file = "/opt/seagate/cortx/sspl/generated_configs"
+        self.cluster_id = ""
+        self.node_id = ""
+        self.rmq_encrypted_passwd = ""
+
+    def _get_result(self, cmd):
+        """ Execute command and return output
+        """
+        result = ""
+        out, _, rc = SimpleProcess(cmd).run()
+        if rc == 0:
+            result = out.decode('utf-8').strip()
+        return result
+
+    def _insert_nested_dict_to_consul(self, v, prefix=''):
+        """ Parse v and insert kv in consul
+        """
+        if isinstance(v, dict):
+            for k, v2 in v.items():
+                p2 = "{}/{}".format(prefix, k)
+                self._insert_nested_dict_to_consul(v2, p2)
+        elif isinstance(v, list):
+            for i, v2 in enumerate(v):
+                p2 = "{}/{}".format(prefix, i)
+                self._insert_nested_dict_to_consul(v2, p2)
+        else:
+            if self.consul_conn is not None:
+                self.consul_conn.kv.put(prefix, str(v))
+            else:
+                print('Consul connection issue, consul_conn is None. Exiting..')
+                sys.exit()
+
+    def set_config(self, component, node, config_file):
+        """ Read config file and insert configs in consul
+        """
+        try:
+            if self.consul_conn is None:
+                raise Exception(f"Failed to establish connection to {CONSUL_HOST}:{CONSUL_PORT}")
+        except Exception as cerror:
+            print("Error in connecting consul | consul: {}".format(cerror))
```

Comment:
We need to end hard consul dependency as here conf in file is enough. Also eventually as config needs will be served through ConfStore, which sspl will integrate with in coming sprints.